tests/ui_tools/test_boxes.py

```diff
@@ -68,6 +68,53 @@ def test_not_calling_send_private_message_without_recipients(self, key,
 
         assert not write_box.model.send_private_message.called
 
+    @pytest.mark.parametrize('key', keys_for_command('SAVE_AS_DRAFT'))
+    @pytest.mark.parametrize('draft_composition', [
+        {
+            'type': 'stream',
+            'to': 'Current Stream',
+            'content': 'Random message',
+            'subject': 'Topic'
+        },
+        {
+            'type': 'private',
+            'to': ['emailgateway@zulip.com'],
+            'content': 'Random message'
+        },
+    ])
+    @pytest.mark.parametrize('draft_saved_in_current_session', [
+        True, False,
+    ], ids=[
+        'saved_draft_in_current_session',
+        'no_saved_draft_in_current_session',
+    ])
+    def test_keypress_SAVE_DRAFT_MESSAGE(self, key, mocker, draft_composition,
+                                         draft_saved_in_current_session,
+                                         write_box, widget_size):
+        mocker.patch(BOXES + '.WriteBox.update_recipient_emails')
+        (write_box.model.session_draft_message.
+            return_value) = draft_saved_in_current_session
+        write_box.stream_write_box = mocker.Mock(edit_text="Current Stream")
+        write_box.title_write_box = mocker.Mock(edit_text="Topic")
+        write_box.msg_write_box = mocker.Mock(edit_text="Random message")
+        write_box.recipient_emails = ['emailgateway@zulip.com']
```

Comment:
Looks like you aren't using the data from test fixture, and still providing a hard-coded value matching with the fixtures - that's why the tests aren't failing. This shouldn't happen; otherwise, there's no reason to use `parametrize`, right :)

Plus, the above three `write_box` objects are only for stream message and the last one for PM's, so probably better to have separate cases; but it's minor.

Reply:
> Looks like you aren't using the data from test fixture, and still providing a hard-coded value matching with the fixtures - that's why the tests aren't failing. This shouldn't happen; otherwise, there's no reason to use parametrize, right 

Yeah my bad. :)